### PR TITLE
[MM-23535] Adds disable database search config flag to the admin console

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Summary
Adds a config flag to disable the database search to the admin console

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-23535
https://mattermost.atlassian.net/browse/MM-23956

#### Related Pull Requests
- [Has server changes](https://github.com/mattermost/mattermost-server/pull/14245)
- [Has enterprise changes](https://github.com/mattermost/enterprise/pull/608)

### CODE REVIEW

**Comment by mgdelacroix** [comment]
For some reason the CI has decided to build a branch with a previous name that I updated and removed prior to create this PR. 

image

[The CI workflow for the PR](https://circleci.com/workflow-run/7b98bb68-4ec0-443e-b908-e7489d9ad4f5) is green and passing, so we can ignore the red check.

**Comment by lindy65** [comment]
Thanks @mgdelacroix - I will test here once the server and enterprise PRs are merged 👍 

Please ping me when ready to QA :)

File: components/admin_console/database_settings.jsx [hunk 1]
```diff
@@ -273,6 +275,24 @@ export default class DatabaseSettings extends AdminSettings {
...
+                    label={
+                        <FormattedMessage
+                            id='admin.sql.disableDatabaseSearchTitle'
+                            defaultMessage='Disable database search: '
+                        />
+                    }
+                    helpText={
+                        <FormattedMessage
+                            id='admin.sql.disableDatabaseSearchDescription'
+                            defaultMessage='Disables the use of the database to perform searches. Should only be used  when other search engines are configured.'
```
Comment (wiersgallak) [inline components/admin_console/database_settings.jsx:289]: Extra space in 2nd sentence between "used" and "when"

File: i18n/en.json [hunk 2]
```diff
@@ -1530,6 +1530,8 @@
   "admin.sql.connMaxLifetimeTitle": "Maximum Connection Lifetime:",
   "admin.sql.dataSource": "Data Source:",
   "admin.sql.dataSourceDescription": "Set the database source in the config.json file.",
+  "admin.sql.disableDatabaseSearchDescription": "Disables the use of the database to perform searches. Should only be used  when other search engines are configured.",
```
Comment (wiersgallak) [inline i18n/en.json:1533]: Extra space in 2nd sentence between "used" and "when" here as well.

**Pull request merged**